testing/e2e/app/routes/Table/spec.ts

```diff
@@ -379,3 +377,66 @@ test.describe('Table row selection', () => {
   };
   //#endregion
 });
+test.describe('Virtual Scroll Tests', () => {
+  test('should render only a few elements out of a big data set', async ({
+    page,
+  }) => {
+    await page.goto('/Table?virtualization=true', { waitUntil: 'networkidle' }); //Need to wait until the virtual rows are able to be rendered for the tests to work.
+
+    const rows = page.getByRole('rowgroup').getByRole('row');
+    expect((await rows.all()).length).toBe(11);
+    await expect(rows.nth(0)).toContainText('Name0');
+    await expect(rows.nth(10)).toContainText('Name10');
+
+    //scroll a little
+    await page.mouse.move(100, 300);
+    await page.mouse.wheel(0, 620);
+    await expect(rows.nth(0)).toContainText('Name10');
+    await expect(rows.nth(10)).toContainText('Name20');
+    expect((await rows.all()).length).toBe(11);
+
+    //scroll back up
+    await page.mouse.wheel(0, -620);
+    await expect(rows.nth(0)).toContainText('Name0');
+    await expect(rows.nth(10)).toContainText('Name10');
+    expect((await rows.all()).length).toBe(11);
+
+    //scroll to end
+    await page.mouse.wheel(0, 6200000);
+    await expect(rows.nth(0)).toContainText('Name99990');
+    await expect(rows.nth(9)).toContainText('Name99999');
+    expect((await rows.all()).length).toBe(10);
+  });
+
+  test('should not crash with empty data objects', async ({ page }) => {
+    await page.goto('/Table?virtualization=true&empty=true', {
+      waitUntil: 'networkidle',
+    }); //Need to wait until the virtual rows are able to be rendered for the tests to work.
+
+    const rows = page.getByRole('rowgroup').getByRole('row');
+    expect((await rows.all()).length).toBe(0);
+  });
+
+  test('virtualized table should scroll to provided row', async ({ page }) => {
+    await page.goto('/Table?virtualization=true&scroll=true', {
+      waitUntil: 'networkidle',
+    }); //Need to wait until the virtual rows are able to be rendered for the tests to work.
+
+    const rows = page.getByRole('rowgroup').getByRole('row');
+    const row50NameCell = page.getByText('Name50');
+    expect((await rows.all()).length).toBe(12);
+    await expect(rows.nth(0)).toContainText('Name44');
+    await expect(rows.nth(6)).toContainText('Name50');
+    await expect(rows.nth(11)).toContainText('Name55');
+
+    await expect(row50NameCell).toBeVisible();
```

Comment:
I believe [`toBeInViewport`](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-in-viewport) is more appropriate here. `toBeVisible` would match even when the row is not actually scrolled to.

Reply:
Replaced with `toBeInViewport` in this [commit](https://github.com/iTwin/iTwinUI/pull/2092/commits/f97de88987b126fea5adcdae5ae2c736389711af).